src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -47,6 +47,225 @@ end
 _symmetric_back(ΔΩ) = UpperTriangular(ΔΩ) + LowerTriangular(ΔΩ)' - Diagonal(ΔΩ)
 _symmetric_back(ΔΩ::Union{Diagonal,UpperTriangular}) = ΔΩ
 
+#####
+##### `Symmetric{<:Real}`/`Hermitian` eigendecomposition
+#####
+
+function frule((_, ΔA), ::typeof(eigen), A::LinearAlgebra.RealHermSymComplexHerm)
+    F = eigen(A)
+    ∂F = Thunk() do
+        λ, U = F
+        M = U' * ΔA * U
+        ∂λ = real(diag(M)) # if ΔA is Hermitian, so is U′ΔAU, so its diagonal is real
+        # K is skew-hermitian with zero diag
+        K = M ./ _nonzero.(λ' .- λ)
+        _setdiag!(K, Zero())
+        ∂U = U * K
+        return Composite{typeof(F)}(values = ∂λ, vectors = ∂U)
+    end
+    return F, ∂F
+end
+
+function rrule(::typeof(eigen), A::LinearAlgebra.RealHermSymComplexHerm)
+    F = eigen(A)
+    function eigen_pullback(ΔF)
```

Comment:
We could do this, but I think it's best to avoid typing for reasons other than dispatch. Unless this is necessary to make sure something else works correctly?